src/Collection.js

```diff
@@ -367,7 +367,7 @@ Collection.prototype.mCreateDocument = function (documents, options, cb) {
   });
 
   self.kuzzle.query(this.buildQueryArgs('document', 'mCreate'), data, options, cb && function (err, res) {
-    cb(err, err ? undefined : res.result);
+    cb(err, res && res.result);
```

Comment:
This will invoke the callback with a `false` result part instead of a `undefined` one in case of a non-partial error.

Fix: `cb(err, res ? res. result : undefined);`